f5/bigip/resource.py

```diff
@@ -295,6 +305,37 @@ def _handle_requests_params(self, kwargs):
             requests_params.update({'params': params})
         return requests_params
 
+    def _check_exclusive_parameters(self, **kwargs):
+        """Check for mutually exclusive attributes in kwargs.
+
+        :raises ExclusiveAttributesPresent
+        """
+        if len(self._meta_data['exclusive_attributes']) > 0:
+            attr_set = set(kwargs.keys())
+            ex_set = set(self._meta_data['exclusive_attributes'][0])
+            common_set = attr_set.intersection(ex_set)
+            if len(common_set) > 1:
+                cset = ', '.join(common_set)
+                error = 'Mutually exclusive arguments submitted. ' \
+                        'The following arguments cannot be set ' \
+                        'together: "%s".' % cset
+                raise ExclusiveAttributesPresent(error)
+
+    @staticmethod
+    def _check_required_parameters(rqset, **kwargs):
+        """Helper function to do operation on sets
+
+        ::returns bool and variable
+        """
+        key_set = set(kwargs.keys())
+        required_minus_received = rqset - key_set
+        if required_minus_received != set():
```

Comment:
The method here would raise MissingRequiredParams(msg) and the caller could then reraise that or raise their own exception with a customized message. My hangup here is that the meaning of return values of True and False may not be explicitly understood by a user of this method. The other thing is that in one case a tuple is returned and in the other, a bool is returned. Raising is a clear indication that something went wrong here, then the caller can wrap that in a try except block and decide to do something based on whether it raised or not.